ESP8266/ESP8266.cpp

```diff
@@ -123,6 +124,10 @@ const char *ESP8266::getIPAddress(void)
         return 0;
     }
 
+    if(std::strncmp(_ip_buffer, "0.0.0.0", sizeof(*_ip_buffer)) == 0) {
```

Comment:
Rather pointless strncmp - if you've overrun your _ip_buffer, it's a bit late to be worrying about it now. The `std::` prefix is meaningless - actually, it's wrong, as you've included `<string.h>`, not `<cstring>`. `std::strncmp` may not provided by `<string.h>`.

Reply:
Rather pointless strncmp - if you've overrun your _ip_buffer, it's a bit late to be worrying about it now
"0.0.0.0" is not a usable IP address - that's the point here.

    The std:: prefix is meaningless - actually, it's wrong, as you've included <string.h>, not <cstring>. std::strncmp may not provided by <string.h>
oops, I should fix that one.

Reply:
I was querying the use of strncmp, as opposed to strcmp - that sort of defensiveness seems pointless when you're looking at your private data.